spec/PushController.spec.js

```diff
@@ -347,13 +347,15 @@ describe('PushController', () => {
      fail('should not succeed');
      done();
    }).catch(() => {
-     let query = new Parse.Query('_PushStatus');
-     query.find({useMasterKey: true}).then((results) => {
-       expect(results.length).toBe(1);
-       let pushStatus = results[0];
-       expect(pushStatus.get('status')).toBe('failed');
-       done();
-     });
+     setTimeout(() => {
+      let query = new Parse.Query('_PushStatus');
+      query.find({useMasterKey: true}).then((results) => {
+        expect(results.length).toBe(1);
+        let pushStatus = results[0];
+        expect(pushStatus.get('status')).toBe('failed');
+        done();
+      });
+     }, 500);
```

Comment:
Need to add some delay here as with DIRECT_ACCESS the status is not updated upon failure...